src/sidebar/components/search-status-bar.js

```diff
@@ -23,41 +27,90 @@ function SearchStatusBarController(store, rootThread) {
     );
   });
 
-  this.filterMatchCount = function() {
-    return visibleCount(thread());
+  const filterResults = () => {
+    const resultsCount = visibleCount(thread());
+    switch (resultsCount) {
+      case 0:
+        return 'No results for ' + filterQuery();
+      case 1:
+        return '1 search result';
+      default:
+        return resultsCount + ' search results';
+    }
   };
 
-  this.areAllAnnotationsVisible = function() {
-    if (store.getState().directLinkedGroupFetchFailed) {
+  const areAllAnnotationsVisible = () => {
+    if (useStore(store => store.getState().directLinkedGroupFetchFailed)) {
       return true;
     }
-    const selection = store.getState().selectedAnnotationMap;
+    const selection = useStore(store => store.getState().selectedAnnotationMap);
     if (!selection) {
       return false;
     }
     return Object.keys(selection).length > 0;
   };
 
-  this.filterQuery = function() {
-    return store.getState().filterQuery;
+  const filterQuery = () => {
```

Comment:
When converting helper methods on controllers from Anguar => React/Preact, any methods that don't take arguments can be replaced by values, ie:

```js
const filterQuery = useStore(store => store.getState().filterQuery)
const filterActive = !!filterQuery
```

In Angular, a method was needed because a component's controller is a class which is instantiated once and the methods are called whenever the data might have changed. In React however the component is a function which is called each time the component is rendered.